rm/rm-node-updater/src/main/java/org/ow2/proactive/resourcemanager/updater/RMNodeUpdater.java

```diff
@@ -179,54 +190,70 @@ private long getRemoteLastModifiedMillis(String url) {
         }
     }
 
+    /**
+     * Check if the node jar is up-to-date and download it
+     * @return true if the local jar is up-to-date or has been successfully renewed. false in case of error
+     */
     private boolean makeNodeUpToDate() {
-        if (!isLocalJarUpToDate(nodeJarUrl, nodeJarSaveAs)) {
-            logger.info("Downloading node.jar from " + nodeJarUrl + " to " + nodeJarSaveAs);
-
-            try {
+        switch (isLocalJarUpToDate(nodeJarUrl, nodeJarSaveAs)) {
+            case OUT_DATED:
+                logger.info("Downloading node.jar from " + nodeJarUrl + " to " + nodeJarSaveAs);
                 File destination = new File(nodeJarSaveAs);
                 File lockFile = null;
                 FileLock lock = null;
+                FileChannel channel = null;
+                try {
+                    if (destination.exists()) {
 
-                if (destination.exists()) {
+                        lockFile = new File(StandardSystemProperty.JAVA_IO_TMPDIR.value(), "lock");
+                        if (!lockFile.exists()) {
+                            lockFile.createNewFile();
+                        }
 
-                    lockFile = new File(StandardSystemProperty.JAVA_IO_TMPDIR.value(), "lock");
-                    if (!lockFile.exists()) {
-                        lockFile.createNewFile();
-                    }
+                        logger.info("Getting the lock on " + lockFile.getAbsoluteFile());
+                        channel = new RandomAccessFile(lockFile, "rw").getChannel();
```

Comment:
![BLOCKER](https://sonarsource.github.io/sonar-github/severity-blocker.png 'Severity: BLOCKER') Use try-with-resources or close this "RandomAccessFile" in a "finally" clause. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2095)